_sources/Unit2-Using-Objects/topic-2-5-methods-return.rst

```diff
@@ -53,21 +53,57 @@
      </svg> Time estimate: 45 min.
    </div>
 
-Calling Methods that Return Values
-===================================
-
-If a method is a **void method** and has **void** as its return type, like most of the methods we have seen so far, that means that it does not return anything. But some methods **return** a value back that the program can use.
-
-Get Methods
-------------
-
-Get methods return the value of instance variables, for example ``getWidth`` and ``getHeight`` to get the height and width for a ``Turtle`` object.
-
-In the previous lesson, we used some set methods with parameters to set the attributes of a turtle to different values, for example yertle.setColor(Color.red); or yertle.setWidth(50); Programmers create get and set methods for each attribute represented as an instance variable in a class to access and modify the value in that variable. The get methods always return back the value of that instance variable, and the set methods modify the value.
-
-When you use a get method, you need to assign what it returns in a variable or use the value in some way for example by printing it out. The data type of the variable must match the data type of the return value of the method. You can find out the return type of a method in its documentation. It will be right before the method name, for example ``int getWidth()`` means ``getWidth`` will return an ``int`` (an integer number).
-
-Here are some examples of using get methods for the ``Turtle`` object ``yertle``.
+Methods that Return Values
+==========================
+
+All the methods on ``Turtle`` that we’ve discussed so far have had a ``void``
+return type. Such methods are sometimes called **void methods**. Because a
+``void`` method don’t return any value, the only point of calling one is because
+it has some **side effect**—it changes the state of the object or maybe causes
+something to happen like drawing a line on the screen. Or both.
+
+In contrast methods with a return type of anything `other` than ``void`` are
+called **non-void** methods. These methods **return** a value that the code
+calling the method can use. And because methods are invoked on an object, these
+methods can be used to return values that tell us things about an object’s
+internal state.
+
+In well-designed programs, non-void methods typically don’t have side effects.
+And void methods obviously can’t return values. So most methods are of one kind
+or the other: either a void method which is called for some side effect or a
+non-void method that is called to compute a value but otherwise has no effect.
+To put it another way, void methods `do things` while non-void methods `produce
+values`.
+
+
+Accessors, a.k.a. Getters
```

Comment:
/ instead of a.k.a. which some kids might not know?

Reply:
Wait, what? The LOL, OMG, WTFBBQ generation doesn't know a.k.a. Dang. Okay, I'll change it.